library/lib/application.cpp

```diff
@@ -132,6 +132,76 @@ static void windowKeyCallback(GLFWwindow* window, int key, int scancode, int act
     }
 }
 
+static void updatePointerPos(float xpos, float ypos){
+    auto& last_cursor_pos{ Application::getLastPointerPos() };
+    auto& current_cursor_pos{ Application::getCurrPointerPos() };
+    auto& delta_cursor_pos{ Application::getDeltaPointerPos() };
+
+    last_cursor_pos = current_cursor_pos;
+    current_cursor_pos.x = xpos;
+    current_cursor_pos.y = ypos;
+    delta_cursor_pos.x = current_cursor_pos.x - last_cursor_pos.x;
+    delta_cursor_pos.y = current_cursor_pos.y - last_cursor_pos.y;
+}
+
+/**
+ * PC Mouse cursor.
+ */
+static void cursorPosCallback(GLFWwindow* window, double xpos, double ypos){
+    updatePointerPos(xpos, ypos);
+
+    if(glfwGetMouseButton(window, GLFW_MOUSE_BUTTON_LEFT) == GLFW_PRESS
+    && (Application::getDeltaPointerPos().x || Application::getDeltaPointerPos().y)
+    ){
+        TouchEvent touch_event;
+        touch_event.type = TouchEventType::DRAG;
+        touch_event.pos = Application::getCurrPointerPos();
+        touch_event.delta = Application::getDeltaPointerPos();
+        // Set the touch event.
+        Application::getTouchEvent() = std::move(touch_event);
```

Comment:
I'm not against std::move but uh can we maybe use more standard OOP stuff (setTouchEvent)? I don't like assigning something to a method return value directly like that

Reply:
Ok I'll do that.